src/ipc/ipc4/handler.c

```diff
@@ -971,8 +971,11 @@ void ipc_cmd(struct ipc_cmd_hdr *_hdr)
 	enum ipc4_message_target target;
 	int err;
 
-	if (!in)
+	if (!in) {
```

Comment:
What is the run-time scenario when the passed pointer is NULL? This code adds more run-time error handling so please clarify (looks like logic error == assert to me, not a real run-time path atm).